### PR TITLE
Sdes 1711:Bulk Download should return status OK instead of Not_Found

### PR DESCRIPTION
**Pull request by ashwinirudrappa**
Modified code to return status as OK when connector.listAvailableFiles returns empty list

### CODE REVIEW

File: test/uk/gov/hmrc/sdes/bulkdownload/controllers/BulkDownloadControllerSpec.scala [hunk 1]
```diff
@@ -44,11 +47,26 @@ class BulkDownloadControllerSpec extends UnitSpec with MockitoSugar {
...
   "BulkDownloadController" should {
-    "call connector with the given fileType" in new Setup {
+    "call connector with the given fileType when data not found" in new Setup {
       when(mockSdesListFilesConnector.listAvailableFiles(any[String])(any[HeaderCarrier]))
         .thenReturn(Future.successful(Nil))
 
-      await(controller.list(fileType)(validRequest))
+      private val result = await(controller.list(fileType)(validRequest))
+      status(result) shouldBe Status.OK
+      contentAsString(result) shouldBe ""
```
Comment (MihailsC-hmrc) [inline test/uk/gov/hmrc/sdes/bulkdownload/controllers/BulkDownloadControllerSpec.scala]:
```suggestion
      contentAsJson(result) shouldBe Json.array()
```

File: test/uk/gov/hmrc/sdes/bulkdownload/controllers/BulkDownloadControllerSpec.scala [hunk 2]
```diff
@@ -44,11 +47,26 @@ class BulkDownloadControllerSpec extends UnitSpec with MockitoSugar {
...
+
+    "call connector with the given fileType" in new Setup {
+      val fileItems = List(FileItem("name", "Url", 10))
+
+      when(mockSdesListFilesConnector.listAvailableFiles(any[String])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(fileItems))
+
+      private val result = await(controller.list(fileType)(validRequest))
+      status(result) shouldBe Status.OK
+      contentAsString(result) shouldBe Json.toJson(fileItems).toString()
```
Comment (MihailsC-hmrc) [inline test/uk/gov/hmrc/sdes/bulkdownload/controllers/BulkDownloadControllerSpec.scala]:
```suggestion
      contentAsJson(result)(result) shouldBe Json.toJson(fileItems)
```

**Review by MihailsC-hmrc — CHANGES_REQUESTED** [review]
We should return `OK` with an empty json list in the response body. Please rework.

File: app/uk/gov/hmrc/sdes/bulkdownload/controllers/BulkDownloadController.scala [hunk 3]
```diff
@@ -42,7 +42,7 @@ class BulkDownloadController @Inject()(sdesListFilesConnector: SdesListFilesConn
     Logger.debug(s"request headers from /list: ${request.headers.toSimpleMap}")
     Logger.debug(s"HeaderCarrier headers from /list: ${hc.headers}")
     sdesListFilesConnector.listAvailableFiles(fileType)(hc) map {
-      case Nil => NotFound
+      case Nil => Ok(JsArray())
       case nonEmptyList => Ok(Json.toJson(nonEmptyList))
     } recover {
```
Comment (MihailsC-hmrc) [inline app/uk/gov/hmrc/sdes/bulkdownload/controllers/BulkDownloadController.scala:46]: Optional sugestion: I propose to return just a list as it is from search results, like
```Scala
sdesListFilesConnector.listAvailableFiles(fileType)(hc)
  .map(list => Ok(Json.toJson(list)))
  .recover {
    ...
  }
```

File: test/uk/gov/hmrc/sdes/bulkdownload/acceptance/BulkDownloadListSpec.scala [hunk 4]
```diff
@@ -78,12 +78,13 @@ class BulkDownloadListSpec extends PlaySpec with GuiceOneAppPerSuite with WireMo
 
 
     "no files are found i.e. SDES proxy returns an empty array" should {
-      "respond with 404 NotFound" in {
+      "respond with 200 OK with empty content" in {
```
Comment (MihailsC-hmrc) [inline test/uk/gov/hmrc/sdes/bulkdownload/acceptance/BulkDownloadListSpec.scala]: Optional: `an empty list` sounds more feasible than `empty content`

**Review by MihailsC-hmrc — APPROVED** [review]

File: test/uk/gov/hmrc/sdes/bulkdownload/acceptance/BulkDownloadListSpec.scala (hunk 4 above)
Comment (ashwinirudrappa) [reply test/uk/gov/hmrc/sdes/bulkdownload/acceptance/BulkDownloadListSpec.scala]: incorporated all the comments

File: app/uk/gov/hmrc/sdes/bulkdownload/controllers/BulkDownloadController.scala (hunk 3 above)
Comment (ashwinirudrappa) [reply app/uk/gov/hmrc/sdes/bulkdownload/controllers/BulkDownloadController.scala:46]: incorporated all the comments

**Review by MihailsC-hmrc — APPROVED** [review]

**Pull request merged**